lib/rack/request.rb

```diff
@@ -234,17 +234,19 @@ def xhr?
 
       def host_with_port
         port = self.port
-        port.nil? || port == DEFAULT_PORTS[scheme] ? host : "#{host}:#{port}"
+        if port.nil? || port == DEFAULT_PORTS[scheme]
+          host
+        else
+          host = self.host
+          # If host is IPv6
+          host = "[#{host}]" if host.include?('::')
```

Comment:
This appear wrong, as you can have valid IPv6 addresses without `::`.  If the host contains any colons at all, it should be considered an IPv6 address.

Reply:
Thank you, changed to `:`.